src/react/HomePage.js

```diff
@@ -29,185 +29,166 @@ import {
 
-  syncData() {
-    this.astarte.getDevicesStats()
-      .then(this.updateDeviceStats)
-      .catch((e) => this.handleSectionError("connectedDevices", e));
+  useEffect(() => {
+    if (!queryAstarteHealth) {
+      setStatus("err");
+      return;
+    }
+
+    let ignore = false;
 
-    this.astarte.getInterfaceNames()
-      .then(this.updateInterfaceNames)
-      .catch((e) => this.handleSectionError("interfaces", e));
+    const getEndpointHealth = async () => {
+      try {
+        const response = await queryAstarteHealth();
+        if (!ignore) {
+          setStatus("ok");
+        }
 
-    this.astarte.getTriggerNames()
-      .then(this.updateTriggerNames)
-      .catch((e) => this.handleSectionError("triggers", e));
+      } catch (err) {
+        setStatus("err");
+      }
+    };
 
-    this.astarte.getRealmManagementHealth()
-      .then(() => { this.updateStatus("realmManagementStatus", "ok") })
-      .catch(() => { this.updateStatus("realmManagementStatus", "err") });
+    getEndpointHealth();
+    return (() => { ignore = true; });
+  }, [queryAstarteHealth]);
 
-    this.astarte.getAppengineHealth()
-      .then(() => { this.updateStatus("appengineStatus", "ok") })
-      .catch(() => { this.updateStatus("appengineStatus", "err") });
+  return status;
+}
 
-    this.astarte.getPairingHealth()
-      .then(() => { this.updateStatus("pairingStatus", "ok") })
-      .catch(() => { this.updateStatus("pairingStatus", "err") });
+const useAstarteAPI = (fetchData) => {
+  const [data, setData] = useState({});
+  const [status, setStatus] = useState("loading");
+  const [error, setError] = useState({});
 
-    if (this.queryFlowHealth) {
-      this.astarte.getFlowHealth()
-        .then(() => { this.updateStatus("flowStatus", "ok") })
-        .catch(() => { this.updateStatus("flowStatus", "err") });
+  useEffect(() => {
+    if (!fetchData) {
+      setStatus("err");
+      setError({ message: "Invalid fetch method" });
+      return;
     }
-  }
 
-  updateDeviceStats(response) {
-    this.setState({
-      connectedDevices: response.data.connected_devices,
-      totalDevices: response.data.total_devices
-    });
-  }
+    let ignore = false;
 
-  updateInterfaceNames(response) {
-    this.setState({
-      interfaces: response.data
-    });
-  }
+    const getData = async () => {
+      try {
+        const response = await fetchData();
+        if (!ignore) {
+          setData(response.data);
+          setStatus("ok");
+        }
 
-  updateTriggerNames(response) {
-    this.setState({
-      triggers: response.data
-    });
-  }
+      } catch (err) {
+        setError(err);
+        setStatus("err");
 
-  handleSectionError(section, err) {
-    this.setState((state) => {
-      return (state[section] === "loading" ? {[section]: "err"} : state)
-    });
+      }
+    };
+
+    getData();
+    return (() => { ignore = true; });
+  }, [fetchData]);
+
+  return [ data, status, error ];
```

Comment:
I suggest to return an object instead of this array, since it could be more easily pattern matched